### ISSUE TITLE
[NativeAOT] string[] not assignable to IList<object>

### ISSUE THREAD

**Issue by jkotas** [dotnet/runtime#106297]
Compiler and run with PublishAot=true:
```
Console.WriteLine(typeof(string[]).IsAssignableTo(typeof(IList<object>)));
```

Actual result: false

Expected result: true

**Comment by jkotas** [comment]
(Same underlying problem as https://github.com/dotnet/runtime/issues/106282. The impact on native AOT is worse.)

**Comment by MichalStrehovsky** [comment]
Do you have the exact repro where `string[]` was not castable to `IList<object>`? I tried your line with both Debug and Release and it prints true.

We even have a test for this exact combination:

https://github.com/dotnet/runtime/blob/3ae95db6cc3e1859ff5517e213457c001dd633f8/src/coreclr/tools/aot/ILCompiler.TypeSystem.Tests/CastingTests.cs#L124

**Comment by am11** [comment]
It might've been tested with net8 AOT?
net9 -> True
net8 -> False

**Comment by MichalStrehovsky** [comment]
Yes, looks like I fixed it in #91788, Jan reviewed it, and then both of us repressed that memory.
